### PR TITLE
Progress

### PR DESCRIPTION
**Pull request by NatalyaTretyak**
Implementation charts for students.

### CODE REVIEW

File: src/app/progress/progress/progress.component.ts [hunk 1]
```diff
@@ -15,15 +15,14 @@ import { TeachersService } from 'src/app/services/teachers.service';
   styleUrls: ['./progress.component.scss']
 })
 export class ProgressComponent implements OnInit {
-
   years: SelectItem[];
   groups: Group[];
   subjects: Subject[];
   classID: number;
-  selectedGroup: any;
+  selectedGroup: Group; // selectedGroup: Group[];
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: this comment is not meaningful

File: src/app/progress/progress/progress.component.ts [hunk 2]
```diff
@@ -38,45 +37,134 @@ export class ProgressComponent implements OnInit {
...
+      this.years = uniqueYears.map(year => ({
+        label: year.toString(),
+        value: year
+      }));
+    });
+  }
+
+  formatDate(date) {
+    const d = new Date(date);
+    // const year = d.getFullYear();
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: comment

File: src/app/progress/progress/progress.component.ts [hunk 3]
```diff
@@ -38,45 +37,134 @@ export class ProgressComponent implements OnInit {
...
+      labels: [],
+      datasets: []
+    };
+
+    const chartData = {
+      labels: [],
+      datasets: []
+    };
+
+    // console.log(this.selectedStudents);
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: comment

File: src/app/progress/progress/progress.component.ts [hunk 4]
```diff
@@ -38,45 +37,134 @@ export class ProgressComponent implements OnInit {
...
+      this.marksService
+        .getMarks(
+          startStr,
+          endStr,
+          this.selectedSubjects.subjectId,
+          this.selectedGroup.id,
+          item.id
+        )
+        .subscribe(data => {
+          // console.log(data);
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: comment

File: src/app/progress/progress/progress.component.ts [hunk 5]
```diff
@@ -38,45 +37,134 @@ export class ProgressComponent implements OnInit {
...
+          /*chartMarksLocal.datasets.push({
+            label: item.lastname + ' ' + item.firstname,
+            data: this.studentMarks,
+            fill: false,
+            borderColor: '#4bc0c0'
+          });*/
+        });
+      console.log(item);
     });
+    /*this.marksService
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: comment

File: src/app/progress/progress/progress.component.ts [hunk 6]
```diff
@@ -38,45 +37,134 @@ export class ProgressComponent implements OnInit {
...
+            backgroundColor: '#7CB342'
+          });
+
+          this.chartMarks = {
+            labels: chartData.labels.map(date => this.formatDate(date)),
+            datasets: chartData.datasets
+          };
+          console.log(this.chartMarks);
+
+          //this.chartMarks.labels = this.markDate;
```
Comment (dmitryy90) [inline src/app/progress/progress/progress.component.ts]: comment

File: src/app/progress/progress/progress.component.ts (hunk 1 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

File: src/app/progress/progress/progress.component.ts (hunk 2 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

File: src/app/progress/progress/progress.component.ts (hunk 3 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

File: src/app/progress/progress/progress.component.ts (hunk 4 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

File: src/app/progress/progress/progress.component.ts (hunk 5 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

File: src/app/progress/progress/progress.component.ts (hunk 6 above)
Comment (NatalyaTretyak) [reply src/app/progress/progress/progress.component.ts]: deleted comment

**Pull request merged**